nyaa/templates/layout.html

```diff
@@ -44,8 +44,174 @@
 		<![endif]-->
 	</head>
 	<body>
+		{% set nyaa_cats = [

```

Comment:
I have the exact same feeling! It is pretty awkward to see these hardcoded into the templates. Moving it server-side can decouple the markup from the database. When there is a new category, we dun need to go through all the codes and apply the change manually. 

Even the `title` column is not needed as, currently, the title of the main categories are merely their name and those of the subcategories are a combination of the name of their respective parent category and their own name with "` - `" in between. It can be constructed on templates, unless this rule is just a coincident and may not be upheld in the future.